plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php

```diff
@@ -262,8 +262,21 @@ protected function getUpdatedFileSize (KalturaDropFolderFile $dropFolderFile)
 	protected function retrieveUserId ($creatorInfo)
 	{
 		$userInfo = $creatorInfo[MicrosoftGraphFieldNames::USER];
-		if(isset($userInfo[MicrosoftGraphFieldNames::EMAIL]))
-		{
+
+		if (isset($userInfo[MicrosoftGraphFieldNames::EMAIL])) {
+			// find user by email and return id if found
+			$filter = new KalturaUserFilter();
+			$filter->emailLike = $userInfo[MicrosoftGraphFieldNames::EMAIL];
```

Comment:
@vberezkin-kaltura I think $filter->emailStartsWith is probably safer to use here.

Reply:
fixed